pyomo/contrib/parmest/utils/model_utils.py

```diff
@@ -94,13 +102,9 @@ def convert_params_to_vars(model, param_names=None, fix_vars=False):
             theta_var_cuid = ComponentUID(theta_object.name)
             theta_var_object = theta_var_cuid.find_component_on(model)
             comp_map[theta_object] = theta_var_object
-            var_theta_objects = []
-            for theta_obj in theta_var_object:
-                theta_cuid = ComponentUID(
-                    theta_var_object.name + '[' + str(theta_obj) + ']'
-                )
-                var_theta_objects.append(theta_cuid.find_component_on(model))
-
+            var_theta_objects = [
+                var_theta_obj for _, var_theta_obj in theta_var_object.items()
+            ]
```

Comment:
If you are only using the values from a dict, it is more clear / more efficient to use `values()`:
```python
var_theta_objects = list(var_theta_obj.values())
```

Reply:
Changed as suggested.